checker/src/main/java/alluxio/checker/SparkIntegrationChecker.java

```diff
@@ -0,0 +1,243 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.checker;
+
+import alluxio.PropertyKey;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.SparkConf;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import scala.Serializable;
+import scala.Tuple2;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * A spark job to test the integration of Spark with Alluxio. This class will be triggered
+ * in yarn-client and yarn-cluster mode through spark-submit.
+ *
+ * This job requires a running Spark cluster, but not requires a running Alluxio cluster.
+ * It will check whether Alluxio classes and Alluxio filesystem can be recognized in Spark
+ * driver and executors.
+ */
+public class SparkIntegrationChecker implements Serializable{
+
+  private static final long serialVersionUID = 1L;
```

Comment:
mvn compile findbugs:findbugs require me to set serialVersionUID when implements Serializable.

Reply:
yepp, this is a common pattern in our code base.
but please use a random number for `serialVersionUID`. I think Intellij has a plugin to help generate it , there might be some online resources too